sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/disbursement.controller.saveDisbursementSchedules.e2e-spec.ts

```diff
@@ -720,38 +724,249 @@ describe("DisbursementController(e2e)-saveDisbursementSchedules", () => {
+          disbursedAmountSubtracted: 50,
+          effectiveAmount: 750,
+        }),
+        createFakeDisbursementValue(
+          DisbursementValueType.CanadaGrant,
+          "CSGP",
+          1500,
+          { effectiveAmount: 1500 },
+        ),
+      ],
+    });
+    firstSchedule.disbursementScheduleStatus = DisbursementScheduleStatus.Sent;
+    // Original assessment - second disbursement (Pending).
+    const secondSchedule = createFakeDisbursementSchedule({
+      disbursementValues: [
+        createFakeDisbursementValue(
+          DisbursementValueType.CanadaGrant,
+          "CSLF",
+          1000,
+        ),
+        createFakeDisbursementValue(DisbursementValueType.BCLoan, "BCSL", 500),
+      ],
+    });
+    secondSchedule.disbursementScheduleStatus =
+      DisbursementScheduleStatus.Pending;
+    fakeOriginalAssessment.disbursementSchedules = [
+      firstSchedule,
+      secondSchedule,
+    ];
+    // Scholastic Standing Change - Withdrawal.
+    const withdrawalOffering = await db.educationProgramOffering.save(
+      createFakeEducationProgramOffering({ auditUser: savedUser }),
     );
-  }
-
-  function assertOveraward(
-    overawards: DisbursementOveraward[],
-    awardCode: string,
-    awardValue: number,
-  ) {
-    expect(overawards).toBeDefined();
-    const awardOverawards = overawards.filter(
-      (overaward) => overaward.disbursementValueCode === awardCode,
+    const scholasticStandingChangeAssessment = createFakeStudentAssessment({
+      auditUser: savedUser,
+      offering: withdrawalOffering,
+      applicationEditStatusUpdatedBy: savedUser,
+    });
+    scholasticStandingChangeAssessment.triggerType =
+      AssessmentTriggerType.ScholasticStandingChange;
+    scholasticStandingChangeAssessment.application = savedApplication;
+    const savedScholasticStandingChangeAssessment =
+      await db.studentAssessment.save(scholasticStandingChangeAssessment);
+    savedApplication.currentAssessment = scholasticStandingChangeAssessment;
+    savedApplication.applicationStatus = ApplicationStatus.Completed;
+    await db.application.save(savedApplication);
+
+    // Act
+    const saveDisbursementSchedulesPayload =
+      createFakeSaveDisbursementSchedulesPayload({
+        assessmentId: savedScholasticStandingChangeAssessment.id,
+        createSecondDisbursement: true,
+      });
+    const saveResult = await disbursementController.saveDisbursementSchedules(
+      saveDisbursementSchedulesPayload,
     );
-    expect(awardOverawards).toHaveLength(1);
-    const [overaward] = awardOverawards;
-    expect(overaward.overawardValue).toBe(awardValue);
-    expect(overaward.originType).toBe(
-      DisbursementOverawardOriginType.ReassessmentOveraward,
+
+    // Asserts
+    expect(saveResult).toHaveProperty(
+      FAKE_WORKER_JOB_RESULT_PROPERTY,
+      MockedZeebeJobResult.Complete,
     );
-  }
+
+    const createdDisbursements = await db.disbursementSchedule.find({
```

Comment:
The query should be adjusted as below to ensure only the required data is returned.
In the current query, `disbursementValues` and `coeUpdatedBy` will be fully selected if not restricted at the `select`.

```ts
const createdDisbursements = await db.disbursementSchedule.find({
  select: {
    id: true,
    coeStatus: true,
    coeUpdatedAt: true,
    coeUpdatedBy: { id: true },
  },
  relations: {
    coeUpdatedBy: true,
  },
  where: {
    studentAssessment: { id: assessmentId },
  },
  loadEagerRelations: false,
});
```